master/buildbot/scripts/start.py

```diff
@@ -129,10 +130,35 @@ def launch(config):
             pidfile.write("{0}".format(proc.pid))
 
 
+def py2Warning(config):
+    if sys.version[0] == '2' and not config['quiet']:
+        print(textwrap.dedent("""\
+
+
+        WARNING: You are running Buildbot with Python 2.7.x !
+        -----------------------------------------------------
+
+        Python 2 is going unmaintained as soon as 2020: https://pythonclock.org/
+
+        To prepare for that transition, we recommend upgrading your buildmaster to run on Python 3.6 now!
+
+        Buildbot open source project is as well deprecating Python 2 for better maintainability.
+
+
+        Buildbot 2.0 going to be released in February 2019 will remove support for Python < 3.5
+        https://github.com/buildbot/buildbot/issues/4439
+
+        On most installation, switching to python3 is just about running the tool 2to3 over the master.cfg.
```

Comment:
On most installationS.

Over the master.cfg *file*.

Reply:
I would rephrase it to drop the "just". In general I try to avoid it when writing documentation because it can discourage people that don't fit in that use case but don't realize it.

> On most installations, switching to Python 3 can be accomplished by running the 2to3 tool over the master.cfg file.